CI/Dockerfile

```diff
@@ -34,63 +57,148 @@ RUN apt-get -y update; \
-RUN /root/etc/CI/docker/build_hdf5.sh
+RUN mkdir -p ${hdf5_build_dir}/build && \
+    cd ${hdf5_build_dir} && \
+    wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-${HDF5_VERSION_major}/hdf5-${HDF5_VERSION}/src/hdf5-${HDF5_VERSION}.tar.gz && \
+    tar -xzf hdf5-${HDF5_VERSION}.tar.gz && \
+    cd build && \
+    ../hdf5-${HDF5_VERSION}/configure --enable-shared \
+        --prefix=${hdf5_install_dir} \
+        CXX=${CXX} \
+        CC=${CC} && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${hdf5_build_dir}
+
 
 FROM hdf5 AS moab
 
-ENV PYTHONPATH=/root/build_dir/moab/bld/pymoab/lib/python3.6/site-packages
+# accessing gloabl ARGs in build stage
+ARG install_dir
+ARG MOAB_BRANCH
+ARG build_dir
+ARG ci_jobs
+ARG CXX
+ARG CC
 
 # Set MOAB env variable
 ENV moab_build_dir=${build_dir}/moab
 ENV moab_install_dir=${install_dir}/moab
 
-ARG MOAB=5.3.0
-ENV MOAB_VERSION ${MOAB}
-#TODO move sh file contents into this Dockerfile
-RUN if [ "${MOAB_VERSION}" != "master" ] && [ "${MOAB_VERSION}" != "develop" ]; then \
-        /root/etc/CI/docker/build_moab.sh; \
-    fi;
+# Set the hdf5 install dir as this is also needed by moab compile
+ENV hdf5_install_dir=${install_dir}/hdf5
+
+RUN mkdir -p ${moab_build_dir}/build && \
+    cd ${moab_build_dir} && \
+    git clone -b ${MOAB_BRANCH} --depth 1 https://bitbucket.org/fathomteam/moab && \
+    cd build && \
+    # static building of moab is also an option and was required for older versions
+    # cmake ../moab -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
+    #               -DCMAKE_INSTALL_PREFIX=${moab_install_dir} \
+    #               -DCMAKE_CXX_COMPILER=${CXX} \
+    #               -DCMAKE_C_COMPILER=${CC} \
+    #               -DENABLE_BLASLAPACK=OFF \
+    #               -DBUILD_SHARED_LIBS=OFF \
+    #               -DENABLE_FORTRAN=OFF && \
+    # make -j${ci_jobs} && \
+    # make install && \
+    # rm -rf * && \
+    cmake ../moab -DCMAKE_INSTALL_RPATH=${hdf5_install_dir}/lib:${moab_install_dir}/lib \
+                  -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_install_dir} \
+                  -DCMAKE_INSTALL_PREFIX=${moab_install_dir} \
+                  -DCMAKE_CXX_COMPILER=${CXX} \
+                  -DCMAKE_C_COMPILER=${CC} \
+                  -DENABLE_BLASLAPACK=OFF \
+                  -DBUILD_SHARED_LIBS=ON \
+                  -DENABLE_FORTRAN=OFF \
+                  -DENABLE_PYMOAB=ON && \
+    # ensure that the installation directory for python packages exist
+    for d in $(echo $PYTHONPATH | tr : '\n'); do mkdir -p $d; done && \
+    make -j${ci_jobs} && \
+    make install && \
+    cd && \
+    rm -rf ${moab_build_dir}
+
+ENV dagmc_build_dir=${build_dir}/DAGMC
+ENV dagmc_install_dir=${install_dir}/DAGMC
+
+ENV dagmc_build_dir_shared=${dagmc_build_dir}-shared
+ENV dagmc_install_dir_shared=${dagmc_install_dir}-shared
+
+ENV dagmc_build_dir_static=${dagmc_build_dir}-static
+ENV dagmc_install_dir_static=${dagmc_install_dir}-static
```

Comment:
```suggestion
```
This was in the original install scripts when they were moved into the docker file but it doesn't look like they are needed or used anywhere

Reply:
Looks like these env are just used in the CI scripts. Seeing as we are moving those into the dockerfile and I can't see anywhere else they are used I'm going to go ahead and remove them.
```
grep -r dagmc_install_dir_static .
./DAGMC/CI/env.sh:export dagmc_install_dir_static=${dagmc_install_dir}-static
./DAGMC/CI/scripts/tests.sh:PATH=${dagmc_install_dir_static}/bin:$PATH make test
./DAGMC/CI/scripts/build_dagmc.sh:    local install_dir=${dagmc_install_dir_static}
```